### PR TITLE
Feature/ced 1174 create generic search method

### PR DESCRIPTION
**Pull request by doron-cedar**

### CODE REVIEW

File: CoreAutomation/Helpers/apiService.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+    private async getHeaders(): Promise<{ [key: string]: string }> {
+        const token = await this.getAuthToken();
+        return {
+            'Content-Type': 'application/json',
+            'x-token': token,
+        };
+    }
+
+    async apiSearch<T>(endpoint: string, payload: any): Promise<T> {
```
Comment (avivm) [inline CoreAutomation/Helpers/apiService.ts:17]: The payload should not be of type `any`. There is a specific payload for that request. Please check the types I sent you.
```
export interface DynamicModelFilter {
	start?: number;
	limit?: number;
	orders: OrderDTO[];
	filterFields: FilterField[];
}
```

Comment (avivm) [reply CoreAutomation/Helpers/apiService.ts:17]: The result return value should be:
```
export interface PagedResult<T> {
	start?: number;
	limit?: number;
	total: number;
	hasMore: boolean;
	results: T[];
}
```

File: CoreAutomation/Helpers/UserApiService.ts [hunk 2]
```diff
@@ -115,82 +110,42 @@ class UserApiService {
...
-            if (response.data.results && response.data.results.length > 0) {
-                const businessId = response.data.results[0].business.id;
-                if (businessId) {
-                    return businessId;
-                } else {
-                    throw new Error('Business ID not found in the response');
-                }
-            } else {
-                throw new Error('No merchant user found');
+        const response = await this.apiService.apiSearch<{ results: any[] }>('/userBusinessRel/search', payload);
```
Comment (avivm) [inline CoreAutomation/Helpers/UserApiService.ts:113]: We should prefer to not use `any` type where possible. We know the type of the userBusinessRel:
```
export interface UserBusinessRelRO extends AbstractJpaRO {
	business?: BusinessRO;
	user?: UserDisplayRO;
	role?: UserRole;
	status?: UserBusinessRelStatus;
	receivesEmails?: boolean;
	receivesWhatsappRates?: boolean;
}
```

**Review by avivm — CHANGES_REQUESTED** [review]
It's a good start. Please check my comments. The types should be improved. It will make the API easier and more comfortable to use.

File: CoreAutomation/Helpers/apiService.ts (hunk 1 above)
Comment (doron-cedar) [reply CoreAutomation/Helpers/apiService.ts]: Fixed

File: CoreAutomation/Helpers/UserApiService.ts (hunk 2 above)
Comment (doron-cedar) [reply CoreAutomation/Helpers/UserApiService.ts]: Fixed

File: CoreAutomation/Helpers/UserApiService.ts [hunk 3]
```diff
@@ -1,11 +1,42 @@
...
-import path from 'path';
-import { CRM_USERNAME, CRM_PASSWORD, CRM_LOGIN, CRM_API_URL, validateRequiredEnvVariables } from '../Utilities/envConfig';
-import { formatBigIntStringToCurrencyString } from '../Utilities/CurrencyUtils';
+import ApiService, { DynamicModelFilter, PagedResult } from './ApiService';
+import { CRM_USERNAME, CRM_PASSWORD, CRM_LOGIN, validateRequiredEnvVariables } from '../../CoreAutomation/Utilities/envConfig';
+import { formatBigIntStringToCurrencyString } from '../../CoreAutomation/Utilities/CurrencyUtils';
 
 validateRequiredEnvVariables(['CRM_USERNAME', 'CRM_PASSWORD', 'CRM_LOGIN']);
 
+interface User {
```
Comment (avivm) [inline CoreAutomation/Helpers/UserApiService.ts:8]: I recommend creating a separate file for the interfaces.

File: CoreAutomation/Helpers/apiService.ts [hunk 4]
```diff
@@ -1,9 +1,39 @@
 import axios, { AxiosError, Method, AxiosRequestConfig } from 'axios';
-import { CRM_API_URL } from '../Utilities/envConfig';
+import { CRM_API_URL } from '../../CoreAutomation/Utilities/envConfig';
 
-class ApiService {
-    private token: string | undefined;
+export interface OrderDTO {
```
Comment (avivm) [inline CoreAutomation/Helpers/apiService.ts:4]: Again, I recommend moving the interfaces for a separate file that holds all the server interfaces.

**Review by avivm — APPROVED** [review]
Looks good. I've added recommendations but it's your decision.

File: CoreAutomation/Helpers/UserApiService.ts (hunk 3 above)
Comment (doron-cedar) [reply CoreAutomation/Helpers/UserApiService.ts]: Fixed

File: file-upload-verification-uncertain-1726401972854.png
Comment (avivm) [inline file-upload-verification-uncertain-1726401972854.png:1]: Is this file committed on purpose?

**Review by avivm — CHANGES_REQUESTED** [review]
Looks good. The only issue is the added image. Is it on purpose?

File: file-upload-verification-uncertain-1726401972854.png
Comment (doron-cedar) [reply file-upload-verification-uncertain-1726401972854.png:1]: No, it's a mistake
I'll fix it

**Review by avivm — APPROVED** [review]

**Pull request merged**